src/http/server.ts

```diff
@@ -758,32 +758,35 @@ export class Server {
   protected getResourceClass(
     request: Drash.Http.Request,
   ): Drash.Interfaces.Resource | undefined {
-    for (const resourceName in this.resources) {
-      const resource = this.resources[resourceName];
-      const pathObjs = resource.paths_parsed!;
-      for (const pathObj of pathObjs) {
-        if (pathObj.og_path === "/" && request.url_path === "/") {
-          return resource;
-        }
-
-        const pathMatchesRequestPathname = request.url_path.match(
-          pathObj.regex_path,
-        );
-        if (pathMatchesRequestPathname == null) {
-          continue;
-        }
-
-        pathMatchesRequestPathname.shift();
-        const pathParamsInKvpForm: { [key: string]: string } = {};
-        pathObj.params.forEach((paramName: string, index: number) => {
-          pathParamsInKvpForm[paramName] = pathMatchesRequestPathname[index];
-        });
+    let resource = undefined;
+    for (const regex in this.paths) {
+      if (regex === "/" && request.url_path === "/") {
+        resource = this.paths[regex];
+        break;
+      }
 
-        request.path_params = pathParamsInKvpForm;
-        return resource;
+      const pathMatchesRequestPathname = request.url_path.match(
+        regex,
+      );
+      if (pathMatchesRequestPathname == null) {
+        continue;
       }
+
+      pathMatchesRequestPathname.shift();
+      resource = this.paths[regex];
+      const pathParamsInKvpForm: { [key: string]: string } = {};
+      resource.paths_parsed.forEach(
+        (pathObj: Drash.Interfaces.ResourcePaths) => {
+          pathObj.params.forEach((paramName: string, index: number) => {
+            pathParamsInKvpForm[paramName] = pathMatchesRequestPathname[index];
+          });
+        },
+      );
+      request.path_params = pathParamsInKvpForm;
+      break;
```

Comment:
This break statement is here because...? Basically we only iterate one time the for loop.

Reply:
we iterate until we find a matching resource. if the resource is found, then we return that resource. if you look further up the method, you'll see see this:

```typescript
      const pathMatchesRequestPathname = request.url_path.match(
        regex,
      );
      if (pathMatchesRequestPathname == null) {
        continue;
      }
```

if the current regex path is not matched with the requested path, then we know that the resource associated with the regex path is not the resource, so we continue and iterate again

Reply:
Hummm... I see.

Reply:
do you think a while loop would be faster? lol...

Reply:
@Guergeiro @ebebbington @saragee3 , i just tried the code using a while loop. not seeing much of a difference. if anything it's a little bit slower. maybe while loop is faster using a larger array?

Reply:
That might be just some weird stuff. In terms of speed, they're both O(n).

Reply:
now using a string as an index. we search the string for indices that are associated with resources. once an index is found, we can use that index to `.get()` on the resource Map.

Reply:
disregard last comment. we're now using the IndexService. see the description in this PR for more information about how that service works. it's documented

Reply:
@Guergeiro, still reviewing?